python_modules/dagster/dagster/_core/definitions/asset_graph_differ.py

```diff
@@ -116,55 +188,122 @@ def from_external_repositories(
         )
 
     def _compare_base_and_branch_assets(
-        self, asset_key: "AssetKey"
-    ) -> Sequence[AssetDefinitionChangeType]:
+        self, asset_key: "AssetKey", include_details: bool = False
```

Comment:
Often, when a method takes a boolean parameter like this, it can be better addressed with composition: https://github.com/dagster-io/internal/discussions/9719. Would that kind of approach make sense here?

Reply:
I suggested going down this route since I think composition would be tricky in this particular case - for each change type, we have to determine (1) if there is a change and (2) optionally, what change was made. There's overlap in work & context between the steps that might be hard to thread if we move to a composition model.

Since there's just one boolean param here, and since this is a purely internal method to deduplicate code between the detail & non-detail paths, I think the parameter/control flow complexity is easy enough to follow, but that's just my 2c.

Reply:
I think I agree with Ben. I think using composition would result in a lot of repetitive code.

Reply:
Got it - not a hill I will die on.